### DISCUSSION TITLE
PURCHASEORDER_REFERENCE_PATTERN - can it be changed to be less strict?

### DISCUSSION THREAD

**Discussion by wafflecrate** [inventree/InvenTree#5983] [Q&A] [answered]
Hello! Still getting to know this awesome app. I'm trying to edit the PURCHASEORDER_REFERENCE_PATTERN entry and am having two issues:

1. My current reference pattern is set to: PO {ref:06d}. So it generates PO 123456. What change do I need to make so there is a dash between the first two numbers so it generates: PO 12-3456?
2. Is there a way to make the PURCHASEORDER_REFERENCE_PATTERN less strict? Because I have a bunch of old PO numbers that follow this format: PO 12-345. But at some point the company ran out of PO numbers and had to add another digit to make it: PO 12-3456. But Inventree will only allow one format or the other. Is it possible to reduce the rigidity or strictness of the PO number reference pattern so it allows users to free type it in without giving you the error "Reference must match required pattern: PO {ref:06d}"? Image attached of the error
![reference must match](https://github.com/inventree/InvenTree/assets/67816847/d17c4545-e5d4-439c-ac86-ded14d470b57)

**Comment by SchrodingersGat** [comment, marked as answer]
You could make the required pattern simply `PO {ref}` and remove the width requirements

**Comment by miggland** [comment]
You can also add `#` or `?` to the pattern, so you have a few options:

| Pattern | Example | Note |
|-------------|--------------|--------|
|PO-12-{ref:04d} | PO-12-3456 | The reference number is 3456, the 12 part is fixed and only "cosmetic" |
|PO-##-{ref:04d} | PO-12-3456 | As above, but two numbers up front can be chosen freely |
|PO-??-{ref:04d} | PO-A1-3456 | Allows other characters as well (also including spaces |
|PO-{ref:04d}?| PO-12345, PO-1234 | You can enter different lengths, but the reference won't include the fifth digit, ie both examples here have the same reference. Would not recommend this.|
